components/eventing-controller/pkg/ems/api/events/types/auth_type.go

```diff
@@ -5,3 +5,11 @@ type AuthType string
 const (
 	AuthTypeClientCredentials AuthType = "oauth2"
 )
+
+func IsInvalidAuthType(value string) bool {
+	return value != string(AuthTypeClientCredentials)
+}
+
+func GetAuthType(_ string) AuthType {
+	return AuthTypeClientCredentials
+}
```

Comment:
Why do we need this function? We can use the already exported const `AuthTypeClientCredentials`.

Reply:
The idea was to keep it extensible for future. If there are more values tomorrow, we pass the string value as a parameter and then return the `AuthType` value